packages/editor/src/hooks/media-upload.js

```diff
@@ -3,9 +3,22 @@
  */
 import { addFilter } from '@wordpress/hooks';
 import { MediaUpload } from '@wordpress/media-utils';
+import { store as coreStore } from '@wordpress/core-data';
+import { useDispatch } from '@wordpress/data';
+
+function MediaUploadWithCacheInvalidation( props ) {
+	const { invalidateResolutionForStoreSelector } = useDispatch( coreStore );
+	const { onClose: originalOnClose, ...rest } = props;
+
+	const onClose = ( ...onCloseArgs ) => {
+		invalidateResolutionForStoreSelector( 'getMediaItems' );
+		originalOnClose?.( ...onCloseArgs );
+	};
+	return <MediaUpload onClose={ onClose } { ...rest } />;
+}
 
```

Comment:
Would we ever want to use `MediaUpload` without the cache invalidation? Wondering if it could be added directly to the MediaUpload component instead of in this wrapper. Or do we want to avoid `media-utils` having a `core-data` dependency?

Reply:
I think the test is that packages that don't depend on a server or WordPress backend like the block editor.

The [media utils package README](https://github.com/WordPress/gutenberg/blob/trunk/packages/media-utils/README.md#L4) however, says the following:

> This package is meant to be used by the WordPress core. It may not work as expected outside WordPress usages.

So might be kosher?

Reply:
I think I lean toward keeping the cache invalidation happening here within the `editor` package as that's where we have the filter to include the `MediaUpload` component. It's within the editor context where we expect the invalidation to happen, so arguably it's up to `editor` to make sure that's the case, whereas possibly `media-utils` shouldn't be too concerned about what's happening outside of itself?

> Wondering if it could be added directly to the MediaUpload component instead of in this wrapper.

My understanding is that the goal of the `MediaUpload` component is to try to be a pretty thin bridge between a React component and the core media library, so I'd probably avoid adding too much in the way of block-editor logic to that package.